File: backend/type/payload/domain.go
package payload

type CreateDnsRecord struct {
	Hostname *string `json:"hostname"`
	DNSType  *string `json:"dnstype"`
	Target   *string `json:"target"`
}

type CreateWebProxy struct {
	Hostname *string `json:"hostname"`
	ServerId *uint64 `json:"server_id"`
	Port     *int    `json:"port"`
}

type EditDnsRecord struct {
	Id       *uint64 `json:"id"`
	Hostname *string `json:"hostname"`
	DNSType  *string `json:"dnstype"`
	Target   *string `json:"target"`
}

type EditWebProxy struct {
	Id       *uint64 `json:"id"`
	Hostname *string `json:"hostname"`
	ServerId *uint64 `json:"server_id"`
	Port     *int    `json:"port"`
}

type DeleteDomainPayload struct {
	Id *int `json:"id"`
}

type Domain struct {
	Id       *uint64       `json:"id"`
	Hostname *string       `json:"hostname"`
	DNSType  *string       `json:"dnstype"`
	Target   *string       `json:"target"`
	Server   *PublicServer `json:"server"`
	Port     *int          `json:"port"`
	Service  *string       `json:"service"`
}

File: frontend/src/types/domain.ts
export interface Domain {
    id: number
    hostname: string
    dnstype: string
    target: string
    server: string
    port: number
}

export interface PublicDomain {
    id: number
    hostname: string
    dnstype: string
    target: string
    server: string
    // port: number
}

export interface DomainResponse {
    success: boolean
    data: Domain
}

export interface ServerListResponse {
    success: boolean
    data: PublicDomain[]
}

export interface CreateDomainPayload {
    hostname: string
    dnstype: string
    target: string
    server_id: number
    port: number
    projectId?: string
    service: string
}
export interface CreateDnsRecord {
    hostname: string
    dnstype: string
    target: string
    projectId?: string
}
export interface CreateWebProxy {
    hostname: string
    server_id: number
    port: number
    projectId?: string
}

